wire-compiler/src/main/java/com/squareup/wire/OutputArtifact.java

```diff
@@ -0,0 +1,44 @@
+package com.squareup.wire;
+
+import java.io.File;
+
+/**
+ * A container class that represents an artifact output from the compiler.
+ */
+public class OutputArtifact {
```

Comment:
I've never seen auto/value before, that's neat, thanks for mentioning it.  Looks like it could save a lot of boiler plate when writing value classes that you put into hash maps and trees.  I'm leaning toward not using it here though for the following reasons:

1) We aren't putting these in hash maps or sorted data structures.  We don't need equals/hashCode/toString at the moment
2) Some of the fields are derived. I'll have to change that up because you have to specify all the fields in the constructor.
3)  Its not being used elsewhere in the code. I'll have to explain why I chose it and its not compelling for above reasons 1 & 2.
4) I'll have to add a new external dependency.